src/Sentry.Diagnostics.DiagnosticSource/Internals/DiagnosticSource/SentryDiagnosticListenerIntegration.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Diagnostics;
+using Sentry.Internal;
+
+namespace Sentry.Internals.DiagnosticSource
+{
+    internal class SentryDiagnosticListenerIntegration : IInternalSdkIntegration
+    {
+        private SentryDiagnosticSubscriber? _subscriber;
+        private IDisposable? _diagnosticListener { get; set; }
+
+        public void Register(IHub hub, SentryOptions options)
+        {
+            _subscriber = new SentryDiagnosticSubscriber(hub, options);
```

Comment:
I think here we should check `options` for something that controls adding this or not